Virtualization MU SLES OpenQA part changes for medium based triggering with new flavors

alice-suse opened the pull request:
This PR mainly achieves https://progress.opensuse.org/issues/160868. Please refer to ticket's description for what are required.
Note: This shall be merged TOGETHER WITH https://progress.opensuse.org/issues/160865. Now need to wait for it.

Explanations for some changes:
- bind all job groups to one unified job group, for reuse, more simplified contents, and wildcard match for products per flavor but different versions
- 5 new flavors created for both TERADATA, ES, and regular products, and bind each of them with specific tests, see https://openqa.suse.de/admin/job_templates/426? and https://openqa.suse.de/admin/job_templates/426?.
- move more product related settings from medium setting to test code, so that we can share the same medium  among all products for the *same* flavor, example:
![image](https://github.com/user-attachments/assets/69c78c6b-7182-4726-adec-e409a1af7109)
- reg code for ltss products are all set in job group yaml `LTSS_REGCODES_SECRET` (new products entering LTSS will need to add the setting into the var)
- reg code and reg email for all non-ltss products is aligned to be `SCC_EMAIL`(vtqe@suse.com) and `SCC_REGCODE`, which is set to be the same  and put in medium (it shall be fine for future products too, because each new SLES product will enable the same SCC_REGCODE during development phase by QA)
- exclude 12sp3 xen tests completely
- for 12sp3/12sp5, on those flavors which have test  qam-[kvm|xen]-install-and-features-test
, remove qam-[kvm|xen]-install-and-sriov-test tests, because sriov is not supported on sle12, so the test mainly does host and guest installation/patches which is already covered in the feature test (still keeps for kernel flavors)
- create a new machine type `64bit-ipmi-12sp5` to run 12sp5 xen tests (because not all machines can run the test successfully on host installation part, the selected machines run well from historical jobs), and bind it to 2 pairs of machines -- worker class live_migrat and live_migrat_5
- to distribute migration tests to multiple pairs of machines for more balanced tests on SUT, assign live_migrat_4(worker class) in tests
- the command to trigger tests in schedule_issue.py, after merging the PR, will be like this(an example): `openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="12-SP3" FLAVOR=Server-DVD-Incidents-VIRT-Kernel-TERADATA BUILD=":888888:kernel-debug" INCIDENT_REPO=xxxx`, nothing else. But for personal triggering purpose, of course, you can specify more settings to customize.
- fix guest_management test failure in 15sp4 non-TD , see failure example http://openqa.qam.suse.cz/tests/78699

## Needles: 
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/-/merge_requests/1671


## Verification run: 

On openqa.qam.suse.cz,
- Job group:  http://openqa.qam.suse.cz/group_overview/174
- Yaml: https://openqa.suse.de/admin/job_templates/426?
    - 15SP5, Flavor Server-DVD-Incidents-VIRT-Tools: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP5&build=%3A888888%3Alibguestfs&groupid=174
    - 15SP4 Flavor: Server-DVD-Incidents-VIRT-Core: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP4&build=%3A888888%3Aqemu&groupid=174
    - 15SP4, Flavor: Server-DVD-Incidents-VIRT-Kernel-TERADATA: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP4&build=%3A888888%3Akernel&groupid=174
    - 15SP3, Flavor: Server-DVD-Incidents-VIRT-Xen: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP3&build=%3A888888%3Axen&groupid=174
    - 12SP5, Flavor: Server-DVD-Incidents-VIRT-Core: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP5&build=%3A888888%3Alibvirt&groupid=174
    - 12SP5, Flavor: Server-DVD-Incidents-VIRT-Kernel: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP5&build=%3A888888%3Akernel-debug&groupid=174
    - 12SP3, Flavor: Server-DVD-Incidents-VIRT-Kernel-TERADATA: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP3&build=%3A999999%3Akernel-debug&groupid=174
  - 15sp4 TERADATA, KERNEL pkg: http://openqa.qam.suse.cz/tests/overview?version=15-SP4&build=%3A000000%3Akernel-default&distri=
  - 15sp4 ltss , libguestfs pkg: http://openqa.qam.suse.cz/tests/overview?distri=sle&build=%3A000000%3Alibguestfs&version=15-SP4
  - 12sp3 TERADATA: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP3&build=%3A000000%3Akernel-default


On openqa.suse.de (vmware & hyperv tests),
- Job group:  https://openqa.suse.de/group_overview/426
- Yaml: https://openqa.suse.de/admin/job_templates/426?
  - 15sp4 TD kernel pkg:
    - https://openqa.suse.de/tests/15348315#details
    - https://openqa.suse.de/tests/15348316#details
  - 15SP4 LTSS open-vm-tools: https://openqa.suse.de/tests/15348318
  - 12sp3(TD) kernel pkg:
    - https://openqa.suse.de/tests/15349525
    - https://openqa.suse.de/tests/15348312

alice-suse commented:
@RoyCai7  @tbaev @martinsmarcelo @Julie-CAO  Welcome to review!

waynechen55 commented on lib/main_common.pm:
```diff
@@ -2435,9 +2435,81 @@ sub load_host_installation_modules {
+    set_var('PATCH_WITH_ZYPPER', 1) unless (check_var('PATCH_WITH_ZYPPER', 0));
+
+    # Set AUTOYAST
+    if (check_var('HOST_INSTALL_AUTOYAST', 1) or check_var('AUTOYAST', 1)) {
```
Why need extra ```HOST_INSTALL_AUTOYAST``` ? I think ```AUTOYAST``` is enough.

alice-suse replied:
That's my understanding too. But `HOST_INSTALL_AUTOYAST` has been in qem settings for long. So keep its usage as it has been.

waynechen55 commented on lib/main_common.pm:
```diff
@@ -2435,9 +2435,81 @@ sub load_host_installation_modules {
+    diag("BUILD is $BUILD, UPDATE_PACKAGE is set to " . get_var('UPDATE_PACKAGE', ''));
+
+    # Set PATCH_WITH_ZYPPER
+    set_var('PATCH_WITH_ZYPPER', 1) unless (check_var('PATCH_WITH_ZYPPER', 0));
```
I think let ```PATCH_WITH_ZYPPER``` is 0 or 1 is enough.

alice-suse replied:
No. This has been discussed, and MU members would like to keep the capability to customize the value based on need.

waynechen55 commented on lib/main_common.pm:
```diff
@@ -2435,9 +2435,81 @@ sub load_host_installation_modules {
+
     if (check_var('ENABLE_HOST_INSTALLATION', 1)) {
         if (get_var('AUTOYAST')) {
             loadtest "autoyast/prepare_profile";
```
It seems that you introduce several new parameters to be set during test run. Is this really necessary ?

alice-suse replied:
I do not introduce any new settings, and just keep their usage like before.

RoyCai7 commented:
Regarding the test suites on http://openqa.qam.suse.cz/group_overview/174, if the test suite is either qam-kvm-install-and-features-test or qam-xen-install-and-features-test, and it's not for 12sp5, it would be advisable to create a new machine specifically for these tests.  name this machine 64bit-ipmi-vft(virtualization feature test).
it is convenient for me to schedule job. HDYT?

alice-suse commented:
> Regarding the test suites on http://openqa.qam.suse.cz/group_overview/174, if the test suite is either qam-kvm-install-and-features-test or qam-xen-install-and-features-test, and it's not for 12sp5, it would be advisable to create a new machine specifically for these tests. name this machine 64bit-ipmi-vft(virtualization feature test). it is convenient for me to schedule job. HDYT?
> 

This job group is for development test at the moment. However it is going to be official group after PRs and reviews done and agreed.

The group is where defines tests for all products (including 12sp5) and all mediums.

> if using 64bit-ipmi, it is not reasonable, because there are a lot of workers using such machine name.

About on which machine(class) to run the tests, the applied principle is that if there is no known problem to run specific test on any single machine by bugs, the tests are good to bind to machine type as general as possible, so that they have chance to test on any machine. This is widely used principle.

RoyCai7 commented:
> > Regarding the test suites on http://openqa.qam.suse.cz/group_overview/174, if the test suite is either qam-kvm-install-and-features-test or qam-xen-install-and-features-test, and it's not for 12sp5, it would be advisable to create a new machine specifically for these tests. name this machine 64bit-ipmi-vft(virtualization feature test). it is convenient for me to schedule job. HDYT?
> 
> This job group is for development test at the moment. However it is going to be official group after PRs and reviews done and agreed.
> 
> The group is where defines tests for all products (including 12sp5) and all mediums, and all non-vmware/hyperv testsuites.
> 
> > if using 64bit-ipmi, it is not reasonable, because there are a lot of workers using such machine name.
> 
> About on which machine(class) to run the tests, the applied principle is that if there is no known problem to run specific test on any single machine by bugs, the tests are good to bind to machine type as general as possible, so that they have chance to test on any machine. This is widely used principle.
> 
> If you worry about the interrupt to official test because it uses potentially any machine to test, actually the devel tests were done only on my reserved machine there. To test if it is fine to make it final binded to the type of machine in official job group, I had to do this. And I think it did not interrupt any scheduled tests, because openqa can tell which worker is busy running jobs and only schedule once job finishes.
> 
> I am not sure if I get your point fully correctly. If needed, we can have offline talk to discuss.

if hit issue, can make change. I will use 64bit-ipmi temporarily.

alice-suse commented:
> > > Regarding the test suites on http://openqa.qam.suse.cz/group_overview/174, if the test suite is either qam-kvm-install-and-features-test or qam-xen-install-and-features-test, and it's not for 12sp5, it would be advisable to create a new machine specifically for these tests. name this machine 64bit-ipmi-vft(virtualization feature test). it is convenient for me to schedule job. HDYT?
> > 
> > 
> > This job group is for development test at the moment. However it is going to be official group after PRs and reviews done and agreed.
> > The group is where defines tests for all products (including 12sp5) and all mediums, and all non-vmware/hyperv testsuites.
> > > if using 64bit-ipmi, it is not reasonable, because there are a lot of workers using such machine name.
> > 
> > 
> > About on which machine(class) to run the tests, the applied principle is that if there is no known problem to run specific test on any single machine by bugs, the tests are good to bind to machine type as general as possible, so that they have chance to test on any machine. This is widely used principle.
> > If you worry about the interrupt to official test because it uses potentially any machine to test, actually the devel tests were done only on my reserved machine there. To test if it is fine to make it final binded to the type of machine in official job group, I had to do this. And I think it did not interrupt any scheduled tests, because openqa can tell which worker is busy running jobs and only schedule once job finishes.
> > I am not sure if I get your point fully correctly. If needed, we can have offline talk to discuss.
> 
> if hit issue, can make change. I will use 64bit-ipmi temporarily.

Sorry, I kind of lost for your situation. You use  64bit-ipmi temporarily for what? If you mean schedule_issue.py change for A5, you do not need to specify machine info at all, in triggering cmd, just as ticket description gives. If you mean official incident triggering, you can continue with before.

alice-suse commented:
Latest validation jobs are updated into description.

alice-suse commented:
I rebased code to latest master and extend code to support 12SP5 LTSS-ES, updated mediums and job group settings, and newly triggered validation runs:

openqa.qam.suse.cz:
- 12SP5 LTSS-ES: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP5&build=%3A654321%3Akernel-default&groupid=174
- 12SP5 LTSS: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP5&build=%3A223456%3Akernel-default&groupid=174
- 15SP4 TERADATA Tools: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP4&build=%3A000000%3Alibguestfs&groupid=174
- 15SP6 Xen: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP6&build=%3A561234%3Axen&groupid=174
- 12SP3 Kernel: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP3&build=%3A456123%3Akernel-default&groupid=174

OSD:
- 12SP5 LTSS: https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=%3A666666%3Aopen-vm-tools&groupid=426
- 12SP5 LTSS ES: https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=%3A333333%3Akernel-default&groupid=426

alice-suse commented:
> I rebased code to latest master and extend code to support 12SP5 LTSS-ES, updated mediums and job group settings, and newly triggered validation runs:
> 
> openqa.qam.suse.cz:
> 
> * 12SP5 LTSS-ES: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP5&build=%3A654321%3Akernel-default&groupid=174
> * 12SP5 LTSS: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP5&build=%3A223456%3Akernel-default&groupid=174
> * 15SP4 TERADATA Tools: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP4&build=%3A000000%3Alibguestfs&groupid=174
> * 15SP6 Xen: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=15-SP6&build=%3A561234%3Axen&groupid=174
> * 12SP3 Kernel: http://openqa.qam.suse.cz/tests/overview?distri=sle&version=12-SP3&build=%3A456123%3Akernel-default&groupid=174
> 
> OSD:
> 
> * 12SP5 LTSS: https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=%3A666666%3Aopen-vm-tools&groupid=426
> * 12SP5 LTSS ES: https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=%3A333333%3Akernel-default&groupid=426

All these verification jobs finished and checked. All expected.

@RoyCai7 @tbaev @martinsmarcelo  Welcome to do final review! Now it supports new flavor for 12sp5 LTSS ES.

tbaev commented:
VR runs and code change looks good to me.

martinsmarcelo commented:
LGTM, but git show on conflict o file main.pm

alice-suse commented:
> LGTM, but git show on conflict o file main.pm

It's caused by merging Julie's PR yesterday. Now resolved the 1 line conflict of importing lib functions.

RoyCai7 commented:
LGTM

alice-suse commented:
Thank you all for the review! Merge it now.
